pkg/kubepkg/kubepkg.go

```diff
@@ -191,169 +241,135 @@ func buildPackage(pkg, arch string, packageDef *PackageDefinition) error {
 			logrus.Info("User-supplied Kubernetes version is a pre-release version")
 			logrus.Info("Setting channel to testing")
-			c.Channel = ChannelTesting
+			bc.Channel = ChannelTesting
 		default:
 			logrus.Info("User-supplied Kubernetes version is a release version")
 			logrus.Info("Setting channel to release")
-			c.Channel = ChannelRelease
+			bc.Channel = ChannelRelease
 		}
 	}
 
-	c.KubernetesVersion, err = getKubernetesVersion(packageDef)
+	bc.KubernetesVersion, err = getKubernetesVersion(packageDef)
 	if err != nil {
 		return errors.Wrap(err, "getting Kubernetes version")
 	}
 
-	c.DownloadLinkBase, err = getDownloadLinkBase(packageDef)
+	bc.DownloadLinkBase, err = getDownloadLinkBase(packageDef)
 	if err != nil {
 		return errors.Wrap(err, "getting Kubernetes download link base")
 	}
 
-	logrus.Infof("Kubernetes download link base: %s", c.DownloadLinkBase)
+	logrus.Infof("Kubernetes download link base: %s", bc.DownloadLinkBase)
 
 	// For cases where a CI build version of Kubernetes is retrieved, replace instances
 	// of "+" with "-", so that we build with a valid Debian package version.
-	c.KubernetesVersion = strings.Replace(c.KubernetesVersion, "+", "-", 1)
+	bc.KubernetesVersion = strings.Replace(bc.KubernetesVersion, "+", "-", 1)
 
-	c.Version, err = getPackageVersion(packageDef)
+	bc.Version, err = getPackageVersion(packageDef)
 	if err != nil {
 		return errors.Wrap(err, "getting package version")
 	}
 
-	logrus.Infof("%s package version: %s", c.Name, c.Version)
+	logrus.Infof("%s package version: %s", bc.Name, bc.Version)
 
-	c.KubeletCNIVersion = minimumCNIVersion
+	bc.KubeletCNIVersion = minimumCNIVersion
 
-	c.Dependencies, err = GetKubeadmDependencies(packageDef)
+	bc.Dependencies, err = GetKubeadmDependencies(packageDef)
 	if err != nil {
 		return errors.Wrap(err, "getting kubeadm dependencies")
 	}
 
-	c.KubeadmKubeletConfigFile = kubeadmConf
+	bc.KubeadmKubeletConfigFile = kubeadmConf
 
-	if c.Arch == "arm" {
-		c.DebArch = "armhf"
-	} else if c.Arch == "ppc64le" {
-		c.DebArch = "ppc64el"
-	} else {
-		c.DebArch = c.Arch
-	}
+	bc.BuildArch = getBuildArch(bc.GoArch, bc.Type)
 
-	c.CNIDownloadLink, err = getCNIDownloadLink(packageDef, c.Arch)
+	bc.CNIDownloadLink, err = getCNIDownloadLink(packageDef, bc.BuildArch)
 	if err != nil {
 		return errors.Wrap(err, "getting CNI download link")
 	}
 
-	logrus.Infof("Building %s package for %s/%s architecture...", c.Package, c.Arch, c.DebArch)
-	return c.run()
+	logrus.Infof("Building %s package for %s/%s architecture...", bc.Package, bc.GoArch, bc.BuildArch)
+	return bc.run()
 }
 
-func (c cfg) run() error {
-	var w []work
-
-	// nolint
-	// TODO: Get package directory for any version once package definitions are broken out
-	srcdir := filepath.Join(latestPackagesDir, c.Package)
-	dstdir, err := ioutil.TempDir(os.TempDir(), "debs")
+func (bc *buildConfig) run() error {
+	workspaceInfo, err := os.Stat(bc.workspace)
 	if err != nil {
 		return err
 	}
```

Comment:
We might want to remove this check because we do an `os.MkdirAll()` in line 308.

Reply:
The `os.Stat()` is primarily here so we can grab `workspaceInfo.Mode()` and use it in `os.MkdirAll` to ensure the file mode is the same.

Reply:
Makes sense :+1: